src/core/services/theming/theming.js

```diff
@@ -528,7 +528,7 @@ function generateAllThemes($injector) {
   if (generateOnDemand) return;
 
   angular.forEach(THEMES, function(theme) {
-    if (!GENERATED[theme.name]) {
+    if (!GENERATED[theme.name] && !(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')) {
```

Comment:
No, that's wrong. Imagine you have set the default theme to `myDefaultTheme`
And it's iterating through the themes, and their is `default` now in progress.

```js
theme.name !== 'default'
// 'default' !== 'default' === FALSE
```
So in this case it works.

But now, if you have no default theme set, then it will never insert the `default` theme and the theming breaks.